core/src/test/java/kafka/testkit/BrokerNode.java

```diff
@@ -121,16 +146,16 @@ public BrokerNode build(
     private final boolean combined;
     private final Map<String, String> propertyOverrides;
 
-    BrokerNode(
+    private BrokerNode(
         Uuid incarnationId,
         MetaPropertiesEnsemble initialMetaPropertiesEnsemble,
         boolean combined,
         Map<String, String> propertyOverrides
     ) {
-        this.incarnationId = incarnationId;
-        this.initialMetaPropertiesEnsemble = initialMetaPropertiesEnsemble;
```

Comment:
`logDataDirectories` should return immutable collection

Reply:
I make `logDataDirectories()` return Set<String> instead List<String> to get rid of the redundant arraylist wrapping.